cylc/flow/parsec/fileparse.py

```diff
@@ -224,6 +229,21 @@ def read_and_proc(fpath, template_vars=None, viewcfg=None, asedit=False):
     do_empy = True
     do_jinja2 = True
     do_contin = True
+
+    if not template_vars:
+        template_vars = {}
+
+    # Load Rose Vars, if a ``rose-suite.conf`` file is present.
+    extra_vars = {
+        'env': None,
+        'empy:suite.rc': None,
+        'jinja2:suite.rc': None
+    }
+    for entry_point in pkg_resources.iter_entry_points(
+        'cylc.pre_configure'
+    ):
+        extra_vars = entry_point.resolve()(Path(fpath).parent)
```

Comment:
If more than one plugin is defined then each plugin that is run will override the results of the previous plugin.

Need to do something like:
```suggestion
        plugin_result = entry_point.resolve()(Path(fpath).parent)
        extra_vars['env'].update(plugin_result.get('env', {}))
```

Reply:
This makes me wonder if you thought about how to handle name clashes. With multiple plugins you might have a situation, where the same variable is being set by multiple plugins, which one wins then? Is the order predictable/controllable? If not, you may potentially end up with weird edge cases. I wonder if the interface shouldn't look something like this:

```python
result -> {
    'template variables': {},
    'template engine': get_template_engine(flines[0]),
    ...
}
for plugin in iter_plugins():
    result = plugin(result)
```

This way each plugin could at least see what the previous ones did and decide what to do (skip, override, merge, throw an error).

Reply:
Fundamentally there are two options:

* Fail if two plugins provide the same variable.
* Override based on a stable order.

We could provide the result to plugins, however, there isn't anything intelligent that the Rose plugin could do to resolve to resolve such an issue.

Reply:
> Fundamentally there are two options:
> 
> * Fail if two plugins provide the same variable.

I would go for that. Can always be revisited later as use cases arise without breaking existing suites.

> * Override based on a stable order.
> 
> We could provide the result to plugins, however, there isn't anything intelligent that the Rose plugin could do to resolve to resolve such an issue.

Fair point, and different plugins might want to implement different and incompatible strategies for conflict resolution.